Stop Image XObjects being included as empty strings in PDFObject::getTextArray().

rupertj opened the pull request:
Fix for #733 and #767.

This PR stops Image XObjects being included in the results of PDFObject:: getTextArray() as empty strings.

k00ni commented:
Thank you for your PR.

Is it still work in progress?

If not, there are a few task left to solve before I consider it. Please read https://github.com/smalot/pdfparser/blob/master/CONTRIBUTING.md for more information.

rupertj commented:
Thanks for pointing me to that @k00ni. I've fixed the code style issues and added test coverage for the change.

I think this is ready to be reviewed.

k00ni commented:
The PR looks good.

I was wondering why it didn't show any CI test results, but found that Microsoft doesn't support Windows 2019 anymore in the workflows (https://github.com/actions/runner-images/issues/12045#issue-2997151424).

#776 should solve that. After it was merged into master, I can give you a hint so you can merge it. Afterwards we should see if the CI has any complains.

k00ni commented:
Please merge in the current master branch to trigger our CI.

daniser commented:
Do this PR also close #761?
